src/Server/TCPServer.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include <Poco/Net/TCPServer.h>
+
+#include <base/types.h>
+#include <Server/TCPServerConnectionFactory.h>
+
+
+namespace DB
+{
+class Context;
+
+class TCPServer : public Poco::Net::TCPServer
+{
+public:
+    explicit TCPServer(
+        TCPServerConnectionFactory::Ptr factory,
+        Poco::ThreadPool & thread_pool,
+        Poco::Net::ServerSocket & socket,
+        Poco::Net::TCPServerParams::Ptr params = new Poco::Net::TCPServerParams);
+
+    /// Close the socket and ask existing connections to stop serving queries
+    void close()
+    {
+        is_open = false;
+        socket.close();
```

Comment:
So the only difference between `TCPServer` and `Poco::Net::TCPServer` is that `TCPServer` closes the socket. Can you add a comment why `Poco::Net::TCPServer::stop()` is not enough and we also need to close the socker?